src/relay/backend/aot_executor_codegen.cc

```diff
@@ -597,8 +647,8 @@ class AOTExecutorCodegen : public MixedModeVisitor {
   runtime::Module* mod_;
   /*! \brief list of input expressions (i.e., variable passed by the user) */
   std::vector<Var> input_vars_;
-  /*! \brief list of device contexts used */
-  std::vector<String> devices_;
+  /*! \brief map of device contexts variables */
+  Map<String, tir::Var> devices_;
```

Comment:
Would it be possible use name_hints of the tir::var to carry the name ? or does it have another purpose ?

Reply:
I use `devices_` in `GenerateAllDeviceHook` and `GenerateDeviceHook` as an argument to the hook functions, so I can keep track of the context vars.